modules/mysql/main.tf

```diff
@@ -22,6 +22,10 @@ locals {
     enabled  = var.ip_configuration
     disabled = {}
   }
+
+  // HA method using REGIONAL availability_type requires binary logs to be enabled
```

Comment:
Is it true that `REGIONAL` `availability_type` requires binary logs? From poking around the docs it seems like binary logs are only required if you want to use a read replica or if you want point in time recovery?

>These settings determine whether automated backups are performed and whether binary logging is enabled or not. Both options add a small performance cost, but are required for the creation of replicas and clones, and for point-in-time recovery.

https://cloud.google.com/sql/docs/mysql/instance-settings#backups-and-binary-logging-2ndgen

Reply:
https://github.com/terraform-google-modules/terraform-google-sql-db/issues/71#issuecomment-604998264

It fails without bin logs enabled, the GCP docs mention this for enabling HA

https://cloud.google.com/sql/docs/mysql/configure-ha#console
https://cloud.google.com/sql/docs/mysql/configure-ha#gcloud

>Leave the Automate backups and Enable point-in-time recovery checkboxes enabled.

Which requires bin logs

Reply:
Very interesting, thanks for those links. I've recreated the same thing for myself with gcloud, the REST API, and the GCP UI.

I'll follow up with a GCP Support ticket to understand if they'll offer more flexibility in the future. Thanks for writing up the PR!